src/main/java/wolox/training/repositories/BookRepository.java

```diff
@@ -1,22 +1,22 @@
 package wolox.training.repositories;
 
-import java.util.List;
+import java.util.Optional;
 import org.springframework.data.repository.CrudRepository;
 import org.springframework.stereotype.Repository;
 import wolox.training.models.Book;
 
-public interface BookRepository extends CrudRepository {
+@Repository
+public interface BookRepository extends CrudRepository<Book, Long> {
 
-	public Book findBookByAuthor(String author);
+	public Optional<Book> findByAuthor(String author);
 
-	public List<Book> findAll();
+	public Iterable<Book> findAll();
 
-	public List<Book> findBookByTitle(String title);
+	public Optional<Iterable<Book>> findByTitle(String title);
```

Comment:
I think this is a mistake, it shouldn't be an optional because since it returns an Iterable, it will return an empty list if there are no books that match. You should either return Optional<Book>, or Iterable<Book> (in which case the name of the method should be findAllByTitle, I think) :)